pkg/azure/blob.go

```diff
@@ -104,6 +105,15 @@ func NewClient(obj *sourcev1.Bucket, secret *corev1.Secret) (c *BlobClient, err
 			c.ServiceClient, err = azblob.NewServiceClientWithSharedKey(obj.Spec.Endpoint, cred, &azblob.ClientOptions{})
 			return
 		}
+
+		var token string
+		if token, err = sasTokenFromSecret(secret); err != nil {
+			return
+		}
+		if token != "" {
+			c.ServiceClient, err = azblob.NewServiceClientWithNoCredential(obj.Spec.Endpoint+token, &azblob.ClientOptions{})
```

Comment:
We need to ensure the resulting URL is always correct. We may also need a test case in which the base URL already contains some query parameters.

Reply:
That's a good point. How would we handle duplicate query parameters? Would we pick the one present in the base URL or from the token?
I am leaning towards the token